ra/1.1/resource-agent-api.md

```diff
@@ -198,90 +244,103 @@ unsupported action the RA shall return an error as defined below.
+    shared memory segments, semaphores, IPC message queues, lock files, etc.
 
     `stop` must succeed if the resource is already stopped.
 
     `stop` must return an error if the resource is not fully stopped.
-  
+
 - `monitor`
 
-    Mandatory.
+    This must check the current status of the resource instance. The
+    thoroughness of the check may optionally be influenced by **Check Levels**.
+    Service must remain available during the monitor, regardless of check
+    level.
 
-    Checks and returns the current status of the resource instance. The
-    thoroughness of the check is further influenced by the weight of the
-    check, which is further explained under **Action specific extensions**..
+- `meta-data`
 
-    It is accepted practice to have additional instance parameters which
-    are not strictly required to identify the resource instance but are
-    needed to monitor it or customize how intrusive this check is allowed
-    to be.
+    This must display the information described under
+    **Resource Agent Meta-Data** via standard output.
 
-    Note that `monitor` shall also return a well defined error code (see
-    below) for stopped instances, ie before `start` has ever been
-    invoked.
-  
-- `recover`
+#### Optional Actions
+
+- `demote`
 
-    Optional.
+    If the resource agent supports roles, this action must put an active
+    resource in the unpromoted role.
+
+- `notify`
+
+    If the resource requires special coordination when multiple instances are
+    run simultaneously in the cluster, the resource agent should support this
+    action, which should perform such coordination.
+
+    When the RA supports this action, RMs should call the action for all active
+    instances of this particular resource in the cluster before and after any
+    demote, promote, start, or stop action performed on any instance of it.
+
+    How the RM passes useful information to the RA when performing this action
+    is currently left to the RM and RA, but may be formalized in a future
+    version of this standard.
+
+- `promote`
+
+    If the resource agent supports roles, this action must put an active
+    resource in the promoted role.
+
+- `recover`
 
     A special case of the `start` action, this should try to recover a resource
-    locally. 
+    locally.
 
     It is recommended that this action is not advertised unless it is
-    advantageous to use when compared to a stop/start operation.
+    advantageous to use when compared to a stop and start action sequence.
 
-    If this is not supported, it may be mapped to a stop/start action by
-    the RM.
+    If this is not supported, it may be mapped to a stop and start action
+    sequence by the RM.
 
     An example includes "recovering" an IP address by moving it to another
     interface; this is much less costly than initiating a full resource group
-    fail over to another node.
+    fail-over to another node.
 
```

Comment:
Pacemaker does not support this action for non-promotable resource. I suppose this is a Pacemaker implementation detail and the standard should not close this door.

So maybe it should be documented somewhere in Pacemaker doc as well?

Reply:
Agreed, that would be worth documenting in Pacemaker. The standard itself makes the recover action optional for agents, and doesn't require resource managers to support any particular actions.

Reply:
ok, thanks.

Reply:
I wanted to create a doc PR, but I'm confused with: 

> One action specified in the OCF specs, recover, is not currently used by the cluster. It is intended to be a variant of the start action that tries to recover a resource locally 

From the following doc:

https://github.com/ClusterLabs/pacemaker/blob/master/doc/sphinx/Pacemaker_Administration/agents.rst

I was pretty it was actually supported?

Reply:
Nope, Pacemaker never supported recover. I'm not aware of any agents that support it, either.

Reply:
ok, I found my mistake, I was mixing up with migrate-to/from actions.

So `recover` is actually documented as not implemented in Pacemaker doc. My bad.

Sorry for your time :(